frontend/packages/shared/src/components/InputActionWrapper/InputActionWrapper.tsx

```diff
@@ -0,0 +1,127 @@
+export type ActionGroup = 'editMode' | 'hoverMode' | 'standBy';
+
+const actionGroupMap: Record<ActionGroup, AvailableAction[]> = {
+  editMode: ['save', 'delete'],
+  hoverMode: ['edit', 'delete'],
+  standBy: [],
+};
+
+const actionToIconMap: Record<AvailableAction, React.ReactNode> = {
+  edit: <PencilWritingIcon />,
+  save: <CheckmarkIcon />,
+  delete: <TrashIcon />,
+};
+
+export type InputActionWrapperProps = {
+  children: React.ReactElement;
+  mode?: ActionGroup;
+  onEditClick: () => void;
+  onDeleteClick: () => void;
+  onSaveClick: () => void;
+};
+
+export const InputActionWrapper = ({
+  mode,
+  children,
+  onEditClick,
+  onDeleteClick,
+  onSaveClick,
+  ...rest
+}: InputActionWrapperProps): JSX.Element => {
+  const t = useText();
+  const [actions, setActions] = useState<AvailableAction[]>(actionGroupMap[mode || 'standBy']);
+
+  const handleFocus = (): void => {
+    setActions(actionGroupMap['editMode']);
+  };
+
+  const handleBlur = (): void => {
+    setActions(actionGroupMap['standBy']);
+  };
+
+  const handleMouseLeave = (): void => {
+    const isEditMode = actions.includes('save');
+    if (isEditMode) return;
+    handleBlur();
+  };
+
+  const handleActionClick = (action: AvailableAction): void => {
+    switch (action) {
+      case 'delete':
+        onDeleteClick();
+        break;
+      case 'save':
+        onSaveClick();
+        handleBlur();
+        break;
+      case 'edit':
+        onEditClick();
+        setActions(actionGroupMap['editMode']);
+      default:
+        break;
+    }
+  };
+
+  const handleHover = (): void => {
+    const isInStandByMode = mode === 'standBy';
+    if (isInStandByMode) {
+      setActions(actionGroupMap['hoverMode']);
+    }
+  };
+
+  // Handled Variant, size, and color in functions if we need to change them based on status  in the future.
+  const actionToAriaLabelMap: Record<AvailableAction, string> = {
+    edit: t('general.edit'),
+    delete: t('general.delete'),
+    save: t('general.save'),
+  };
+
+  const handleVariant = (action: AvailableAction): 'filled' | 'quiet' | 'outline' => {
+    return 'quiet';
+  };
+
+  const handleColor = (action: AvailableAction): 'danger' | 'primary' | 'secondary' | 'success' => {
+    switch (action) {
+      case 'delete':
+        return 'danger';
+      case 'save':
+        return 'success';
+      case 'edit':
+        return 'primary';
+      default:
+        return 'primary';
+    }
+  };
+
+  const handleSize = (action: AvailableAction): 'small' | 'medium' | 'large' => {
+    return 'large';
+  };
+
+  return (
+    <div className={classes.buttons} onMouseOver={handleHover} onMouseLeave={handleMouseLeave}>
+      {React.cloneElement(children, {
+        ...rest,
+        onFocus: handleFocus,
+      })}
+      {actions.map((action) => (
+        <Button
+          variant={handleVariant(action)}
+          size={handleSize(action)}
+          color={handleColor(action)}
```

Comment:
Ref. prev comment, then we do not need to make a function call here but a lookup in our map instead. 😊 
```suggestion
          color={actionToColorMap[action]}
```